scripts/llm/qat_flow.py

```diff
@@ -0,0 +1,204 @@
+{%- else %}
+    {%- set system_message = '' %}
+{%- endif %}
+{{- '<|start_header_id|>system<|end_header_id|>\n\n' }}
+{{- system_message }}
+{{- '<|eot_id|>' }}
+{%- for message in messages %}
+    {%- if message.role == 'assistant' %}
+        {%- generation %}
+            {{- '<|start_header_id|>assistant<|end_header_id|>\n\n' + message['content'] | trim + '<|eot_id|>'}}
+        {%- endgeneration %}
+    {%- else %}
+        {{- '<|start_header_id|>' + message['role'] + '<|end_header_id|>\n\n' + message['content'] | trim + '<|eot_id|>' }}
+    {%- endif %}
+{%- endfor %}
+{%- if add_generation_prompt %}
+    {{- '<|start_header_id|>assistant<|end_header_id|>\n\n' }}
+{%- endif %}
+"""
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        description="NeMo2.0 QAT/QAD simplified flow. Currently supports models that fit on 1 node and 8 GPUs."
+    )
+    QUANT_CFG_CHOICES_LIST = ["no_quant", *get_quant_cfg_choices()]
+
+    parser.add_argument(
+        "--model-path",
+        type=str,
+        help="Path to the NeMo 2.0 checkpoint",
+        required=True,
+    )
+    parser.add_argument(
+        "--finetune-recipe",
+        type=str,
+        default="llama3_8b",
+        help="Choose NeMo 2.0 recipe. Recipes are named in the format of <model_name>_<model_size>(_<long_sequence_length> or other special settings)",
+    )
+    parser.add_argument(
+        "--distill",
+        action="store_true",
+        help="Whether to do quantization aware distillation (QAD)",
+    )
+    parser.add_argument(
+        "--hf-tokenizer",
+        type=str,
+        help="Name of HF model to use for tokenizer.",
+        required=False,
+    )
+    parser.add_argument(
+        "--chat-template",
+        type=str,
+        help="Path to the custom chat template to replace the HF tokenizer default chat template.",
+        required=False,
+    )
+    parser.add_argument(
+        "--data-path",
+        type=str,
+        help="Path to the finetuning chat dataset. Can be either ShareGPT or HuggingFace/OpenAI chat format",
+        required=True,
+    )
+    parser.add_argument(
+        "-algo",
+        "--algorithm",
+        type=str,
+        default="fp8",
+        choices=QUANT_CFG_CHOICES_LIST,
+        help="TensorRT-Model-Optimizer quantization algorithm",
+    )
+    parser.add_argument(
+        "--slurm",
+        action="store_true",
+        help="Run on slurm using run.SlurmExecutor",
+        default=False,
+    )
+    parser.add_argument(
+        "--experiment",
+        type=str,
+        help="Experiment name",
+        default="qat_flow",
+    )
+    return parser
+
+
+def get_finetune_recipe(recipe):
+    assert hasattr(
+        llm, recipe
+    ), f"Recipe named {recipe} not found. General format is <model_name>_<model_size>(_<long_sequence_length> or other special settings)"
+    finetune_recipe = getattr(llm, recipe).finetune_recipe
+    return finetune_recipe(peft_scheme=None)  # TODO add dir
```

Comment:
override the PEFT scheme here since it defaults to lora sometimes in the NeMo recipes